### ISSUE TITLE
Mailable ExtractTags BadMethodCallException

### ISSUE THREAD

**Issue by Jason-Morcos** [laravel/telescope#504]
A job attempts to send an email that extends Illuminate's Mailable and uses Queueable, and SerializesModels. The following crash occurs.

Telescope v1.0.9

```

BadMethodCallException: Method Illuminate\Database\Eloquent\Collection::getKey does not exist. in /vendor/laravel/framework/src/Illuminate/Support/Traits/Macroable.php:100

Stack trace:
#0 /vendor/laravel/telescope/src/FormatModel.php(18): Illuminate\Support\Collection->__call('getKey', Array)
#1 /vendor/laravel/telescope/src/ExtractTags.php(30): Laravel\Telescope\FormatModel::given(Object(Illuminate\Database\Eloquent\Collection))
#2 [internal function]: Laravel\Telescope\ExtractTags::Laravel\Telescope\{closure}(Object(Illuminate\Database\Eloquent\Collection), 1)
#3 /vendor/laravel/framework/src/Illuminate/Support/Collection.php(1007): array_map(Object(Closure), Array, Array)
#4 /vendor/laravel/telescope/src/ExtractTags.php(31): Illuminate\Support\Collection->map(Object(Closure))
#5 /vendor/laravel/telescope/src/ExtractsMailableTags.php(19): Laravel\Telescope\ExtractTags::from(Object(App\Mail\CustomMailable))
#6 [internal function]: Laravel\Telescope\Telescope::Laravel\Telescope\{closure}(Object(App\Mail\ CustomMailable))
#7 /vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(284): call_user_func(Object(Closure), Object(App\Mail\CustomMailable))
#8 /vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(153): Illuminate\Mail\Mailable->buildViewData()
#9 /vendor/laravel/framework/src/Illuminate/Support/Traits/Localizable.php(19): Illuminate\Mail\Mailable->Illuminate\Mail\{closure}()
#10 /vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(160): Illuminate\Mail\Mailable->withLocale(NULL, Object(Closure))
#11 /vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(274): Illuminate\Mail\Mailable->send(Object(Illuminate\Mail\Mailer))
#12 /vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(229): Illuminate\Mail\Mailer->sendMailable(Object(App\Mail\ CustomMailable))

```

**Comment by themsaid** [comment]
Can't replicate this, please share more information and example code.

**Comment by Jason-Morcos** [comment]
We hacked together some tests and it appears the issue actually has to do with the way telescope Extracts Tags.

1. Telescope registers mailable tag tracking despite the MailWatcher being disabled. `\Laravel\Telescope\Telescope.php`:119 Calls `static::registerMailableTagExtractor()` regardless of watcher settings.

2. Stepping through ExtractTags from() method
 - `$target` passed into the `from()` method is of class `App\Mail\SensorBatteryStatusEmail` (this is a mailable).
 - `static:modelsFor([$target])` (called line 29) returns an `Illuminate\Support\Collection`.
 - Mapping through the contents the first `$model` is of class `App\Objects\Emailer\EmailerStatus` (which is expected)
- Mapping through the contents the next `$model` is of class `Illuminate\Database\Eloquent\Collection`. Line 30 then throws a `BadMethodCallException` as FormatModel tries to call `getKey()` on a collection. 
This indicates some sort of strange error in ExtractTags's `modelsFor()` as this seems to be expected to return a Collection of Objects (without sub collections). I believe this may be due to nested collections being passed in the mailable. Below I have included a screenshot of a dd of the Mailable illustrating the nested collections.



**App\Mail\SensorBatteryStatusEmail dd():**
<img width="445" alt="screen shot 2018-12-14 at 9 46 37 am" src="https://user-images.githubusercontent.com/10710367/50010177-ba3c1380-ff86-11e8-9b27-ed858af85d0e.png">




**Full Exception Stack Trace:**
```
#0 /chsh/vendor/laravel/telescope/src/FormatModel.php(18): Illuminate\Support\Collection->__call('getKey', Array)
#1 /chsh/vendor/laravel/telescope/src/ExtractTags.php(30): Laravel\Telescope\FormatModel::given(Object(Illuminate\Database\Eloquent\Collection))
#2 [internal function]: Laravel\Telescope\ExtractTags::Laravel\Telescope\{closure}(Object(Illuminate\Database\Eloquent\Collection), 1)
#3 /chsh/vendor/laravel/framework/src/Illuminate/Support/Collection.php(1007): array_map(Object(Closure), Array, Array)
#4 /chsh/vendor/laravel/telescope/src/ExtractTags.php(31): Illuminate\Support\Collection->map(Object(Closure))
#5 /chsh/vendor/laravel/telescope/src/ExtractsMailableTags.php(19): Laravel\Telescope\ExtractTags::from(Object(App\Mail\SensorBatteryStatusEmail))
#6 [internal function]: Laravel\Telescope\Telescope::Laravel\Telescope\{closure}(Object(App\Mail\SensorBatteryStatusEmail))
#7 /chsh/vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(284): call_user_func(Object(Closure), Object(App\Mail\SensorBatteryStatusEmail))
#8 /chsh/vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(153): Illuminate\Mail\Mailable->buildViewData()
#9 /chsh/vendor/laravel/framework/src/Illuminate/Support/Traits/Localizable.php(19): Illuminate\Mail\Mailable->Illuminate\Mail\{closure}()
#10 /chsh/vendor/laravel/framework/src/Illuminate/Mail/Mailable.php(160): Illuminate\Mail\Mailable->withLocale(NULL, Object(Closure))
#11 /chsh/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(274): Illuminate\Mail\Mailable->send(Object(Illuminate\Mail\Mailer))
#12 /chsh/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(229): Illuminate\Mail\Mailer->sendMailable(Object(App\Mail\SensorBatteryStatusEmail))
#13 /chsh/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(223): Illuminate\Mail\Mailer->send(Object(App\Mail\SensorBatteryStatusEmail))
#14 /chsh/app/Helpers/EmailerHelper.php(197): Illuminate\Support\Facades\Facade::__callStatic('send', Array)
#15 /chsh/app/Helpers/EmailerHelper.php(167): App\Helpers\EmailerHelper::send(Object(Illuminate\Database\Eloquent\Collection), Object(App\Objects\Emailer\EmailerStatus), 'Batteries to Ch...')
#16 /chsh/app/Console/Commands/EmailerExecuteCommand.php(125): App\Helpers\EmailerHelper::sendEscalation(Object(App\Objects\Emailer\EmailerStatus))
#17 /chsh/app/Console/Commands/EmailerExecuteCommand.php(60): App\Console\Commands\EmailerExecuteCommand->handleExistingStatuses()
#18 [internal function]: App\Console\Commands\EmailerExecuteCommand->handle()
#19 /chsh/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#20 /chsh/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#21 /chsh/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#22 /chsh/vendor/laravel/framework/src/Illuminate/Container/Container.php(572): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#23 /chsh/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\Container\Container->call(Array)
#24 /chsh/vendor/symfony/console/Command/Command.php(255): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#25 /chsh/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#26 /chsh/vendor/symfony/console/Application.php(901): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#27 /chsh/vendor/symfony/console/Application.php(262): Symfony\Component\Console\Application->doRunCommand(Object(App\Console\Commands\EmailerExecuteCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#28 /chsh/vendor/symfony/console/Application.php(145): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#29 /chsh/vendor/laravel/framework/src/Illuminate/Console/Application.php(89): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#30 /chsh/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Illuminate\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#31 /chsh/artisan(37): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#32 {main}
```

**Comment by Jason-Morcos** [comment]
Any updates? Was anyone able to replicate or fix this crash?

**Comment by themsaid** [comment]
I believe it's only related to your project as I haven't seen anyone report it and I couldn't replicate.

**Comment by JeremyJass** [comment]
I work with Jason, and looking at the issue a little further think I've identified the root cause, and have a potential fix (PR #529 above).

The problem arises when you take a simple Collection of Models, and execute a 'groupBy' or otherwise modify the Collection so it is no longer single-level. In the ExtractTags file, modelsFor() was assuming that if a property of the target class was an EloquentCollection that it was always a simple Collection of Models.
